packages/sdk-metrics/src/aggregator/exponential-histogram/mapping/LogarithmMapping.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as ieee754 from './ieee754';
+import * as util from '../util';
+import { Mapping, MappingError } from './types';
+
+/**
+ * LogarithmMapping implements exponential mapping functions for scale > 0.
+ * For scales <= 0 the exponent mapping should be used.
+ */
+export class LogarithmMapping implements Mapping {
+  static readonly MIN_SCALE = 1;
+  static readonly MAX_SCALE = 20;
+  private static _PREBUILT_MAPPINGS = new Map<number, LogarithmMapping>();
+
+  /**
+   * get creates or returns a memoized logarithm mapping function for
+   * the given scale. used for scales > 0.
+   * @param scale - a number > 0 and <= 20
+   * @returns {LogarithmMapping}
+   */
+  static get(scale: number): LogarithmMapping {
+    if (
+      scale > LogarithmMapping.MAX_SCALE ||
+      scale < LogarithmMapping.MIN_SCALE
+    ) {
+      throw new MappingError(
+        `logarithm mapping requires scale in the range [${LogarithmMapping.MIN_SCALE}, ${LogarithmMapping.MAX_SCALE}]`
+      );
+    }
+
+    let mapping = this._PREBUILT_MAPPINGS.get(scale);
+    if (mapping) {
+      return mapping;
+    }
+
+    mapping = new LogarithmMapping(scale);
+    this._PREBUILT_MAPPINGS.set(scale, mapping);
+    return mapping;
```

Comment:
optional nit: i prefer this style for these types of behaviors

```suggestion
    if (!this._PREBUILT_MAPPINGS.has(scale)) {
      this._PREBUILT_MAPPINGS.set(scale, new LogarithmMapping(scale));
    }
    return this._PREBUILT_MAPPINGS.get(scale);
```